File: c version that might work/audio-prototype/server/server_windows.cpp
#include <iostream>
#include <portaudio.h>
#include <winsock2.h>

#define SAMPLE_RATE 44100
#define FRAMES_PER_BUFFER 1024
#define PORT 50007

int main() {
    WSADATA wsa;
    WSAStartup(MAKEWORD(2,2), &wsa);

    SOCKET server = socket(AF_INET, SOCK_STREAM, 0);
    sockaddr_in addr {};
    addr.sin_family = AF_INET;
    addr.sin_addr.s_addr = INADDR_ANY;
    addr.sin_port = htons(PORT);

    bind(server, (sockaddr*)&addr, sizeof(addr));
    listen(server, 1);

    std::cout << "Waiting for connection...\n";
    SOCKET client = accept(server, nullptr, nullptr);
    std::cout << "Client connected.\n";

    Pa_Initialize();
    PaStream* stream;
    Pa_OpenDefaultStream(&stream, 0, 1, paInt16, SAMPLE_RATE, FRAMES_PER_BUFFER, nullptr, nullptr);
    Pa_StartStream(stream);

    int16_t buffer[FRAMES_PER_BUFFER];
    while (true) {
        int bytes = recv(client, (char*)buffer, sizeof(buffer), 0);
        if (bytes <= 0) break;
        Pa_WriteStream(stream, buffer, bytes / sizeof(int16_t));
    }

    Pa_StopStream(stream);
    Pa_CloseStream(stream);
    Pa_Terminate();
    closesocket(client);
    closesocket(server);
    WSACleanup();
    return 0;
}

File: failed stuff here/minimal-barrier/audio-prototype/server/server_windows.py
# server.py (Windows)
import socket
import pyaudio

CHUNK = 1024
FORMAT = pyaudio.paInt16
CHANNELS = 1
RATE = 44100
PORT = 50007

p = pyaudio.PyAudio()
stream = p.open(format=FORMAT,
                channels=CHANNELS,
                rate=RATE,
                output=True,
                frames_per_buffer=CHUNK)

s = socket.socket()
s.bind(('0.0.0.0', PORT))
s.listen(1)
print("Waiting for connection...")

conn, addr = s.accept()
print("Connected by", addr)

try:
    while True:
        data = conn.recv(CHUNK * 2)
        if not data:
            break
        stream.write(data)
except KeyboardInterrupt:
    print("Interrupted")

stream.stop_stream()
stream.close()
p.terminate()
conn.close()
s.close()
